examples/customize-toolbar-excel-download/index.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import MUIDataTable from '../../src';
+import CustomToolbar from './CustomToolbar';
+// import XLSX from 'xlsx';
+
+const Example = () => {
+  const [tableState, setTableState] = React.useState();
+
+  const handleXLSXDownload = () => {
+    if (!!tableState) {
+      // manipulate tableState here to your liking, download all data, sorted, filtered, visible...
+      const columns = tableState.columns;
+      const filteredRowIndexArray = tableState.dData.map(row => row.dataIndex);
+
+      const data = tableState.data.filter(row => filteredRowIndexArray.includes(row.index));
+      const visibleColumns = columns.filter(col => col.display === 'true' && col.download === true);
+
+      // boolean array of visible columns
+      const colMap = columns.map(col => col.display === 'true');
+      const filteredData = data.map(({ data }) => data.filter((cell, i) => colMap[i] === true));
+
+      const sheetHeader = [visibleColumns.map(col => col.label)];
+      const data = sheetHeader.concat(filteredData);
```

Comment:
Beware of duplicate `const` declaration here, as this will throw an error.

Reply:
Yep, definitely a bug. Will fix asap. Thanks!